src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/ReturningHandlerChainExceptionBehavior.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Data.SqlClientX.Handlers
+{
+    /// <summary>
+    /// Options for how exceptions should be handled in the <see cref="ReturningHandlerChain{TParameters,TOutput}"/>.
+    /// </summary>
+    internal enum ReturningHandlerChainExceptionBehavior
```

Comment:
Do you have a few usecases in mind for SqlClient, where the ReturningHandler concept can be leveraged?

Reply:
At least a couple -
* Feature extension parsing - feed a feature extension token in as the parameters to a chain of responsibility with a separate handler for each supported feature extension, returning an appropriate feature extension object (or class that inherits from feature extension)
* Row reading - read a token off the stream, pass it into a chain of responsibility with a separate handler for each object type, the one that can handle the object will return the appropriately parsed object.